src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs

```diff
@@ -0,0 +1,141 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.actions.view
+  (:require
+    [quo2.core :as quo]
+    [react-native.core :as rn]
+    [react-native.permissions :as permissions]
+    [react-native.platform :as platform]
+    [react-native.reanimated :as reanimated]
+    [reagent.core :as reagent]
+    [status-im2.common.alert.events :as alert]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]
+    [status-im2.contexts.chat.messages.list.view :as messages.list]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]
+    [status-im2.contexts.chat.bottom-sheet-composer.actions.style :as style]))
+
+(defn send-message
+  [{:keys [input-ref]}
+   {:keys [text-value focused?]}
+   {:keys [height saved-height last-height opacity background-y container-opacity]}
+   window-height]
+  (reanimated/animate height c/input-height)
+  (reanimated/set-shared-value saved-height c/input-height)
+  (reanimated/set-shared-value last-height c/input-height)
+  (reanimated/animate opacity 0)
+  (when-not @focused?
+    (js/setTimeout #(reanimated/animate container-opacity 0.7) 300))
+  (js/setTimeout #(reanimated/set-shared-value background-y
+                                               (- window-height))
+                 300)
+  (rf/dispatch [:chat.ui/send-current-message])
+  (rf/dispatch [:chat.ui/set-input-maximized false])
+  (rf/dispatch [:chat.ui/set-input-content-height c/input-height])
+  (reset! text-value "")
+  (.clear ^js @input-ref)
+  (messages.list/scroll-to-bottom))
+
+(defn send-button
+  [props
+   {:keys [text-value] :as state}
+   animations
+   window-height
+   images?]
+  [:f>
+   (fn []
+     (let [btn-opacity (reanimated/use-shared-value 0)
+           z-index     (reagent/atom 0)]
+       [:f>
```

Comment:
@ulisesmac The effect does not execute in every re-render. It executes only when `max-height` changes.

Reply:
Hmmm... probably you're talking about a different effect?

If you add a `println` in the effect-fn then it will print on every text type:
```clojure
(rn/use-effect (fn []
                 (println "Re executed") ;; <- added here
                 (if (or (not-empty @text-value) images?)
                   (when-not (= @z-index 1)
                     (reset! z-index 1)
                     (js/setTimeout #(reanimated/animate btn-opacity 1) 50))
                   (when-not (= @z-index 0)
                     (reanimated/animate btn-opacity 0)
                     (js/setTimeout #(reset! z-index 0) 300))))
               [@text-value])
```

Probably this effect is not the best to show what is described in [the issue mentioned previously](https://github.com/status-im/status-mobile/issues/15552#issue-1649290578), but this pattern for functional components surely is not the best, also you could take a look at [reagent docs](https://github.com/reagent-project/reagent/blob/master/doc/ReactFeatures.md#function-components).

Talking about the hook itself, I think it's not needed to be watching `@text-value` since we don't want to re-execute this effect each time the user types, I think we only care if the text is empty or not. I tried replacing `[@text-value]` by `[(empty? @text-value)]` and I think it works the same but the hook is not re-executed each time the user types so it's better.

Could you please try this apprach and let me know if it does works as expected?
thank you Omar

Reply:
Thanks @ulisesmac, works well 👍

Reply:
@ulisesmac as for the way of using the functional components, can you please show me an example using the `send-button` component, or any of the other functional components here?